src/articles/FindArticles.js

```diff
@@ -7,9 +7,66 @@ import * as s from "./FindArticles.sc";
 import LoadingAnimation from "../components/LoadingAnimation";
 import { setTitle } from "../assorted/setTitle";
 import strings from "../i18n/definitions";
+import Reminder from "./Reminder";
+import ExtensionMessage from "../components/ExtensionMessage";
+import Feature from "../features/Feature";
+import LocalStorage from "../assorted/LocalStorage";
+import { isMobile } from "../utils/misc/mobileDetection";
+
+/*global chrome*/
+// (this will let our linter know we are accessing Chrome browser methods)
 
 export default function NewArticles({ api }) {
   const [articleList, setArticleList] = useState(null);
+  const [hasExtension, setHasExtension] = useState(false);
+  const [isChrome, setIsChrome] = useState(false);
+  const [extensionMessageOpen, setExtensionMessageOpen] = useState(false);
+  const [displayedExtensionPopup, setDisplayedExtensionPopup] = useState(false);
+
+
+  useEffect(() => {
+    setDisplayedExtensionPopup(LocalStorage.displayedExtensionPopup());
+    if (runningInChromeDesktop()) {
+      setIsChrome(true);
+      if (Feature.extension_experiment1() && !displayedExtensionPopup) {
+        checkExtensionInstalled();
+      }
+    }
+  }, []);
+
+  function runningInChromeDesktop() {
+    let userAgent = navigator.userAgent;
+    return userAgent.match(/chrome|chromium|crios/i) && isMobile() === false;
+  }
+
+  function checkExtensionInstalled() {
```

Comment:
One more reason to try to abstract this further.
Remember the principle of: the name of the method should summarize what it does?

now, if your method tells me that it checksIfTheExtension is installed, why does it also SetExtensionMesageOpen? 

If it were focused on doing one thing, and one thing only, we could implement it like this:

```js

function checkExtensionInstalled(setHasExtension) {
  if (chrome.runtime) {
    chrome.runtime.sendMessage(
      process.env.REACT_APP_EXTENSION_ID,
      "You are on Zeeguu.org!",
      function (response) {
        if (chrome.runtime.lastError) {
          console.log(chrome.runtime.lastError);
        }
        if (response.message === true) {
          setHasExtension(true);
        }
      }
    );
  } else {
    setHasExtension(false);
  }
}
```
and moreover, we could move it out of this file, to say `./util/misc/extensionCommunication.js` 

then we could also move `runningInChromeDesktop` to `util/misc/mobileDetection` (maybe rename that one to be browserDetection to be able to hold both the functions)

This would let our FindArticles to breathe a bit more, and would minimize the amount of functionality that is present in here, that does not have anything to do with FindArticles. 

Finally, it might turn out that we will at a later point want to use `checkExtensionInstalled` also in the Settings page. In that case, we can nicely reuse the function from the `extensionCommunication.js`.

Do you see what I mean? 

It's a bit late for me to figure out where should you move the `setExtensionMessageOpen(true);` but I'm sure you can do it :)

Reply:
I requested a new review. Let me know, if any more changes needs to be done :)